app/src/util.tsx

```diff
@@ -140,6 +140,18 @@ function useArray<T extends unknown[]>(input: T): T {
   }
 }
 
+/**
+ * This custom hook allows using the results of multiple futures. At first, the
+ * initial value is returned. Once all futures are ready, their resolved values
+ * are passed to the `then` function, and the result is returned.
```

Comment:
```suggestion
 * This custom hook allows using the results of multiple promises. At first, the
 * initial value is returned. Once all promises are ready, their resolved values
 * are passed to the `then` function, and the result is returned.
```
uber nit: may avoid confusion--JS doesn't have "futures", and promises behave somewhat distinctly from Rust futures (eagerly vs. lazily evaluated).